Changes the instruction sequence name slightly for the top-most
instruction-sequence container (ISEQ_TYPE_TOP). The name <top
(required)> isn't very instructive and it is not unique. So we replace
(required) with a file name. For example:

 <top /tmp/l.rb> vs <top (required)> 

and also
 
 <RubyVM::InstructionSequence:<compiled>> vs.
 <RubyVM::InstructionSequence:<compiled>@<compiled>>

Motivation: Instruction sequence names become important the more one
needs to access them.  In a debugger we use instruction sequences in
describing where to stop or in showing disassembly, so making things
look more descriptive is helpful.

diff --git a/iseq.c b/iseq.c
index a51a19e..5639c52 100644
--- a/iseq.c
+++ b/iseq.c
@@ -676,9 +676,13 @@ iseq_inspect(VALUE self)
         return rb_sprintf("#<%s: uninitialized>", rb_obj_classname(self));
     }
 
-    return rb_sprintf("<%s:%s@%s>",
-                      rb_obj_classname(self),
-		      RSTRING_PTR(iseq->name), RSTRING_PTR(iseq->filename));
+    if (ISEQ_TYPE_TOP == iseq->type)
+	return rb_sprintf("<%s:%s>",
+			  rb_obj_classname(self), RSTRING_PTR(iseq->name));
+    else
+	return rb_sprintf("<%s:%s@%s>",
+			  rb_obj_classname(self),
+			  RSTRING_PTR(iseq->name), RSTRING_PTR(iseq->filename));
 }
 
 static
diff --git a/load.c b/load.c
index 0ff4b60..4bfc157 100644
--- a/load.c
+++ b/load.c
@@ -8,6 +8,13 @@
 #include "dln.h"
 #include "eval_intern.h"
 
+#ifdef HAVE_SYS_PARAM_H
+# include <sys/param.h>
+#endif
+#ifndef MAXPATHLEN
+# define MAXPATHLEN 1024
+#endif
+
 VALUE ruby_dln_librefs;
 
 #define IS_RBEXT(e) (strcmp((e), ".rb") == 0)
@@ -301,11 +301,13 @@ rb_load_internal(VALUE fname, int wrap)
     if (state == 0) {
 	NODE *node;
 	VALUE iseq;
+	char iseq_name[MAXPATHLEN];
 
 	th->mild_compile_error++;
 	node = (NODE *)rb_load_file(RSTRING_PTR(fname));
 	loaded = TRUE;
-	iseq = rb_iseq_new_top(node, rb_str_new2("<top (required)>"), fname, rb_realpath_internal(Qnil, fname, 1), Qfalse);
+	snprintf(iseq_name, sizeof(iseq_name), "<top %s>", RSTRING_PTR(fname));
+	iseq = rb_iseq_new_top(node, rb_str_new2(iseq_name), fname, rb_realpath_internal(Qnil, fname, 1), Qfalse);
 	th->mild_compile_error--;
 	rb_iseq_eval(iseq);
     }
